app/templates/components/public/session-list.hbs

```diff
@@ -0,0 +1,8 @@
+{{#each-in sessions as |id trackSession|}}
```

Comment:
Now. When it is filtered by a track, what is going to come here is an array and not an object. Please handle that too. 

TL,DR; 
The session-list component should be able to render both an array of sessions and an object (when it is grouped by track) 

You can use the `is-array` helper available